aliyun-ecs/src/main/java/org/jclouds/aliyun/ecs/features/ImageApi.java

```diff
@@ -88,13 +99,19 @@
          }
 
          @Override
-         protected Function<Object, IterableWithMarker<Image>> markerToNextForArg0(final Optional<Object> arg0) {
+         protected Function<Object, IterableWithMarker<Image>> markerToNextForArgs(final List<Object> args) {
+            if (args == null || args.isEmpty()) throw new IllegalStateException("Can't advance the PagedIterable");
+            final String regionId = args.get(0).toString();
+            final Optional<Object> optionalPaginationOptions = Optional.fromNullable(!args.isEmpty() && args.size() == 2 ? args.get(1) : null);
```

Comment:
You could better use the `instanceOf` method from the `Predicates` class for a more safe approach.